### PR TITLE
jthread 1.3.3 (new formula)

### PR DESCRIPTION
**Pull request by njh**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

The main reason for adding jthread to Homebrew, is so that I can also raise a PR for [JRTPLIB](https://github.com/j0r1/JRTPLIB) which depends upon the jthread library.

~~Note that `brew audit --strict` does not pass, because of passing a string with spaces into `system`, which is used in the test. The reason I have done this, is because one of the arguments is itself a string with spaces in it - so must be parsed by the shell.~~

~~I looked at doing this instead:~~
```ruby
libs = `pkg-config --libs jthread`.chomp.split
```

~~But it start getting complex/unreliable compared with just passing a single string to `system`.~~

~~Pointers to a better way of doing this would be very welcome.~~

### CODE REVIEW

File: Formula/jthread.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      int main() {
+        JMutex* jm = new JMutex();
+        jm->Init();
+        jm->Lock();
+        jm->Unlock();
+        return 0;
+      }
+    EOS
+
+    libs = `pkg-config --libs jthread`.chomp
```
Comment (fxcoudert) [inline Formula/jthread.rb]: `pkg-config` is a build dep, so it's not there anymore for the test block. Just expand to the necessary arguments (`-I#{include} -L#{lib} -ljthread` or something like that)

File: Formula/jthread.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    cflags = `pkg-config --cflags jthread`.chomp
+
+    system "#{ENV.cxx} test.cpp -o test #{libs} #{cflags}"
```
Comment (fxcoudert) [inline Formula/jthread.rb]: Once you've removed the temporary variables above, just separate the arguments here. See how `libogg` is doing it.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by njh** [comment]
Thank you very much.

I thought it was good to invoke pkg-config to ensure that it was working correctly.

But specifying the linker parameters directly should work fine.

I will update the PR.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
Thanks @njh for your contribution to Homebrew!

**Pull request closed without merging**